lib/src/test/java/tech/relaycorp/awaladroid/messaging/ReceiveMessagesTest.kt

```diff
@@ -134,6 +135,47 @@ internal class ReceiveMessagesTest : MockContextTestCase() {
         assertTrue(collectParcelsCall.wasCalled)
         assertTrue(messages.isEmpty())
         assertTrue(ackWasCalled)
+        assertTrue(logCaptor.warnLogs.contains("Malformed incoming parcel"))
+    }
+
+    @Test
+    fun receiveParcelWithUnknownRecipient_ackedButNotDeliveredToApp() = runBlockingTest {
+        val channel = createEndpointChannel(RecipientAddressType.PUBLIC)
+        val parcel = buildParcel(channel)
+        var ackWasCalled = false
+        val parcelCollection = parcel.toParcelCollection { ackWasCalled = true }
+        val collectParcelsCall = CollectParcelsCall(Result.success(flowOf(parcelCollection)))
+        pdcClient = MockPDCClient(collectParcelsCall)
+
+        channel.firstPartyEndpoint.delete()
+
+        val messages = subject.receive().toCollection(mutableListOf())
```

Comment:
As you noticed, this is failing with the error: `MissingKeyException: There is no session key for 03d6f89f597e881f275831faaa7abb92553f18fc563e53494372db18461533178`.

That's because this test currently simulates a situation where the 1st party endpoint exists when we start collecting parcels, but no longer exists by the time we get the parcel. However, I think what you meant to simulate is the scenario where the 1st party endpoint didn't even exist when parcel collection started -- that is, `parcel.recipientAddress != channel.firstPartyEndpoint.privateAddress` is what you want to replicate.

I think the former is worth testing to make sure it's handled (I don't think I considered this scenario but it's great we came across it as it could certainly happen in real life!)

I'm not sure we can test the latter, but I could be wrong: If the 1st party endpoint didn't exist when parcel collection started (because it never existed or it was deleted), then that should/will throw an `InvalidMessageException` in:

https://github.com/relaycorp/awala-endpoint-android/blob/bad95859feadbf8f09082c9e3aa3d56854c6d566/lib/src/main/java/tech/relaycorp/awaladroid/messaging/ReceiveMessages.kt#L84-L92

... because the sender's PDA wouldn't have been issued by any id certificate of our 1st party endpoints.

Thoughts?

Reply:
By "I think the former is worth testing to make sure it's handled" I mean add a `catch` for `MissingKeyException`